services/order.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+
+from django.db import transaction
+
+from django.contrib.auth import get_user_model
+
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, User
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: datetime | None = None,
+) -> None:
+    with transaction.atomic():
+        user = User.objects.get(username=username)
+        order = Order.objects.create(
+            user=user)
+        if date:
+            order.created_at = date
+
+        order.save()
```

Comment:
I think it's better to move this save statement to the `if` block so as not to make a query when not needed